moped-editor/src/views/projects/projectView/ProjectComponents/ComponentListItem.js

```diff
@@ -19,6 +20,31 @@ const useStyles = makeStyles((theme) => ({
   },
 }));
 
+const isSignalComponent = (component) =>
+  component?.moped_components?.feature_layer?.internal_table ===
+  "feature_signals";
+
+/**
+ * Format the list item's primary and secondary text
+ * @param {object} component - the moped_component object
+ * @returns {object} the component props for <ListItemText> as { primary, secondary }
+ */
+const useComponentListItemText = (component) =>
+  useMemo(() => {
+    const listItemText = { primary: "", secondary: "" };
+    const componentName = component?.moped_components?.component_name;
+    const componentSubtype = component?.moped_components?.component_subtype;
+    listItemText.primary = componentSubtype
+      ? `${componentName} - ${componentSubtype}`
+      : componentName;
+    if (isSignalComponent(component)) {
+      const signalLocationName = component?.feature_signals?.[0]?.location_name;
+      const signalId = component?.feature_signals?.[0]?.signal_id;
+      listItemText.secondary = `${signalId}: ${signalLocationName}`;
+    }
+    return listItemText;
+  }, [component]);
```

Comment:
should i move this to a separate util module?

Reply:
I moved these helpers to a separate util file in https://github.com/cityofaustin/atd-moped/pull/982